### PR TITLE
build_install_collection - fix install for collection that do not defined version in galaxy.yml

### PR DESCRIPTION
**Pull request by abikouo**
For some Ansible collections `cloud.common`, `vmware.vmware_rest`, etc, the `galaxy.yml` file does not define a version. therefore these collections cant be installed automatically, we need an additional step to build them.
This pull request aims to add extra steps to build and install theses kind of collection

### CODE REVIEW

**Comment by cidrblock** [comment]
Is this needed given this precommit hook exists to automatically update it as needed: https://github.com/ansible-collections/ansible.netcommon/blob/main/.pre-commit-config.yaml#LL6

**Comment by abikouo** [comment]
> Is this needed given this precommit hook exists to automatically update it as needed: https://github.com/ansible-collections/ansible.netcommon/blob/main/.pre-commit-config.yaml#LL6

it is, we are going to use it in a github action context and this is doing more than just updating the version into galaxy.yml

**Comment by cidrblock** [comment]
> > Is this needed given this precommit hook exists to automatically update it as needed: https://github.com/ansible-collections/ansible.netcommon/blob/main/.pre-commit-config.yaml#LL6
> 
> it is, we are going to use it in a github action context and this is doing more than just updating the version into galaxy.yml

If the version in the galaxy.yml file is updated with a precommit hook, the other github actions will only run after the precommit change and push, so it would seem the existing build steps would work because the galaxy.yaml does have a version.

Maybe I misunderstood the root issue here, the PR desc suggests the issue exists because their galaxy.yaml doesn't have a version.... ELI5?

**Comment by abikouo** [comment]
> ELI5

For some collections like `ansible-collections/cloud.common` (https://github.com/ansible-collections/cloud.common/blob/main/galaxy.yml) and `ansible-collections/vmware.vmware_rest` (https://github.com/ansible-collections/vmware.vmware_rest/blob/964773b0f105a37e0e1af273d3f09c5df9931b04/galaxy.yml#L4) the `galaxy.yml` is not committed with a version, so we do not want the precommit to push change on the pull request, therefore we need this job to generate collection version before running integration jobs, hope this clarifies

**Comment by cidrblock** [comment]
I guess then why is the use of precommit to update the version in the galaxy undesirable?  All the certified and validated network collection including ansible.utils and ansible.scm are going this route...  

The more collections doing it the same way the easier things are to maintain since we have a shared repo of actions that are consistently used.

**Comment by gravesm** [comment]
`cloud.common` and `vmware.vmware_rest` are the odd ones out, and as @cidrblock says, the best solution will be to change those collections to bring them in line with everything else.

**Comment by abikouo** [comment]
This is not needed anymore, collection with specificities should be standardized
